app/indexers/rights_metadata_indexer.rb

```diff
@@ -100,10 +42,34 @@ def to_solr
     'http://opendatacommons.org/licenses/odbl/1.0/' => 'odc-odbl'
   }.freeze
 
+  def rights_descriptions
+    return 'controlled digital lending' if cocina.access.controlledDigitalLending
+
+    basic_access = build_basic_access(cocina.access)
+    files = Array(cocina.structural.contains).flat_map { |fs| fs.structural.contains }
+                                             .map { |file| build_basic_access(file.access) }.uniq
+
+    [basic_access] + files.map { |result| "#{result} (file)" }
+  end
+
+  def build_basic_access(access)
+    basic_access = case access.access
+                   when 'citation-only'
+                     return 'citation'
+                   when 'dark'
```

Comment:
Omit dark since will be covered by `else`?

Reply:
This would cause the function to return `dark (no-download)`, which is not the current value (although it is technically correct).

Reply:
Oh right, I see.